### DISCUSSION TITLE
ISR with new pages

### DISCUSSION THREAD

**Discussion by maria-mcparland** [vercel/next.js#39076] [Help]
Hi,
We are running our app using SSG and ISR (deployed on AWS). Today we noticed that pages added to our CMS aren't being picked up properly by the ISR. 
In the nextjs [docs](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) it says '_When a request is made to a path that hasn’t been generated, Next.js will server-render the page on the first request. Future requests will serve the static file from the cache_.'

Does this mean once that page has been stored in cache, ISR will not run on the page again? So if content changes on it, it won't update? 

If that is the case, would on-demand revalidate allow us to pick up new content?

**Comment by leerob** [comment]
How are you deploying - container-based or on a single server? 

https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#self-hosting-isr

**Reply by maria-mcparland** [reply]
Deploying on EKS using docker. 

Just wanting to confirm that new pages should regenerate like pages that were brought in at build time.

That line in the documentation confused us slightly.
